include/deal.II/base/symmetric_tensor.h

```diff
@@ -3311,7 +3311,7 @@ deviator(const SymmetricTensor<2, dim, Number> &t)
   SymmetricTensor<2, dim, Number> tmp = t;
 
   // subtract scaled trace from the diagonal
-  const Number tr = trace(t) / dim;
+  const Number tr = trace(t) / internal::NumberType<Number>::value(dim);
```

Comment:
Since `dim` is a template, I would suggest to write
```suggestion
  const Number tr = trace(t) * internal::NumberType<Number>::value(1.0 / dim);
```
This way, the compiler evaluates the `1/dim` expression at compile time and avoids the division at run time.